.github/actions/upload/src/filecoin.js

```diff
@@ -0,0 +1,262 @@
+import { promises as fs } from 'node:fs'
+import { RPC_URLS, TIME_CONSTANTS } from '@filoz/synapse-sdk'
+import { ethers } from 'ethers'
+import { createCarFromPath } from 'filecoin-pin/dist/add/unixfs-car.js'
+import { validatePaymentSetup } from 'filecoin-pin/dist/common/upload-flow.js'
+import {
+  checkAndSetAllowances,
+  computeTopUpForDuration,
+  depositUSDFC,
+  getPaymentStatus,
+} from 'filecoin-pin/dist/synapse/payments.js'
+import {
+  cleanupSynapseService,
+  createStorageContext,
+  initializeSynapse as initSynapse,
+} from 'filecoin-pin/dist/synapse/service.js'
+import { getDownloadURL, uploadToSynapse } from 'filecoin-pin/dist/synapse/upload.js'
+import { formatRunwayDuration } from 'filecoin-pin/dist/utils/time.js'
```

Comment:
filecoin-pin should have exports map if we're okay with lib-like functionality. see #62